yarn-project/cli/src/utils.ts

```diff
@@ -211,9 +213,58 @@ export function parseAztecAddress(address: string): AztecAddress {
 }
 
 /**
- * Parses a TxHash from a string. Throws InvalidArgumentError if the string is not a valid.
+ * Parses an AztecAddress from a string.
+ * @param address - A serialised Aztec address
+ * @returns An Aztec address
+ * @throws InvalidArgumentError if the input string is not valid.
+ */
+export function parseOptionalAztecAddress(address: string): AztecAddress | undefined {
+  if (!address) {
+    return undefined;
+  }
+  return parseAztecAddress(address);
+}
+
+/**
+ * Parses a selector from a string.
+ * @param selector - A serialised selector.
+ * @returns A selector.
+ * @throws InvalidArgumentError if the input string is not valid.
+ */
+export function parseOptionalSelector(selector: string): FunctionSelector | undefined {
+  if (!selector) {
+    return undefined;
+  }
+  try {
+    return FunctionSelector.fromString(selector);
+  } catch {
+    throw new InvalidArgumentError(`Invalid selector: ${selector}`);
+  }
+}
+
+/**
+ * Parses a string into an integer or returns undefined if the input is falsy.
+ *
+ * @param value - The string to parse into an integer.
+ * @returns The parsed integer, or undefined if the input string is falsy.
+ * @throws If the input is not a valid integer.
+ */
+export function parseOptionalInteger(value: string): number | undefined {
+  if (!value) {
+    return undefined;
+  }
+  const parsed = parseInt(value, 10);
```

Comment:
You probably don't want to use parseInt with a radix, since users may supply numbers in hex:
```
> parseInt('0x20', 10)
0
```

Even without the radix, it's risky if the user drops the leading 0x:
```
> parseInt('20ab')
20
```

I'd just go for `Number`:
```
> Number('0x20')
32
> Number('20')
20
> Number('20ab')
NaN
```

But DYOR to make sure I'm not missing any annoying case.

Reply:
Good catch. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/2600/commits/3228808ae0e0912ae5e927b8370a5222306b9556